bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/handler/NikoHomeControlAlarmHandler.java

```diff
@@ -0,0 +1,217 @@
+        NhcAlarm nhcAlarm = nhcComm.getAlarmDevices().get(deviceId);
+        if (nhcAlarm == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.configuration-error.deviceId");
+            return;
+        }
+
+        nhcAlarm.setEventHandler(this);
+
+        updateProperties(nhcAlarm);
+
+        String location = nhcAlarm.getLocation();
+        if (thing.getLocation() == null) {
+            thing.setLocation(location);
+        }
+
+        this.nhcAlarm = nhcAlarm;
+
+        initialized = true;
+        deviceInitialized();
+    }
+
+    @Override
+    void refresh() {
+        NhcAlarm alarm = nhcAlarm;
+        if (alarm != null) {
+            alarmEvent(alarm.getState());
+        }
+    }
+
+    @Override
+    public void dispose() {
+        NikoHomeControlCommunication nhcComm = getCommunication(getBridgeHandler());
+        if (nhcComm != null) {
+            NhcAlarm alarm = nhcComm.getAlarmDevices().get(deviceId);
+            if (alarm != null) {
+                alarm.unsetEventHandler();
+            }
+        }
+        nhcAlarm = null;
+        super.dispose();
+    }
+
+    private void updateProperties(NhcAlarm nhcAlarm) {
+        Map<String, String> properties = new HashMap<>();
+
+        if (nhcAlarm instanceof NhcAlarm2) {
+            NhcAlarm2 alarm = (NhcAlarm2) nhcAlarm;
+            properties.put(PROPERTY_DEVICE_TYPE, alarm.getDeviceType());
+            properties.put(PROPERTY_DEVICE_TECHNOLOGY, alarm.getDeviceTechnology());
+            properties.put(PROPERTY_DEVICE_MODEL, alarm.getDeviceModel());
+        }
+
+        thing.setProperties(properties);
+    }
+
+    @Override
+    public void alarmEvent(String state) {
+        NhcAlarm nhcAlarm = this.nhcAlarm;
+        if (nhcAlarm == null) {
+            logger.debug(" alarm device with ID {} not initialized", deviceId);
+            return;
+        }
+
+        updateState(CHANNEL_CONTROL,
+                NHCOFF.equals(state) || NHCDETECTORPROBLEM.equals(state) ? OnOffType.OFF : OnOffType.ON);
+        updateState(CHANNEL_ARMED,
+                NHCOFF.equals(state) || NHCPREARMED.equals(state) || NHCDETECTORPROBLEM.equals(state) ? OnOffType.OFF
+                        : OnOffType.ON);
+        updateState(CHANNEL_STATE, StringType.valueOf(ALARMSTATES.get(state)));
+        updateState(CHANNEL_TARGET_STATE,
+                NHCOFF.equals(state) || NHCDETECTORPROBLEM.equals(state) ? StringType.valueOf(ALARMSTATES.get(NHCOFF))
+                        : StringType.valueOf(ALARMSTATES.get(NHCARMED)));
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void alarmTriggerEvent() {
+        NhcAlarm nhcAlarm = this.nhcAlarm;
+        if (nhcAlarm == null) {
+            logger.debug(" alarm device with ID {} not initialized", deviceId);
+            return;
+        }
+
+        triggerChannel(CHANNEL_ALARM);
+        updateStatus(ThingStatus.ONLINE);
```

Comment:
I believe we had this discussion about setting thing status when triggering a channel. I do not remember the conclusion.
So just check that it is consistent.

Reply:
Indeed, and I kept it, as this is coming from the NHC system, and receiving something effectively means the handler is online, even when it didn't report back before.